src/nlp/arrow_dataset.py

```diff
@@ -689,25 +689,87 @@ def map_function(batch, *args):
         # return map function
         return self.map(map_function, batched=True, with_indices=with_indices, arrow_schema=arrow_schema, **kwargs)
 
+
+    def index(
```

Comment:
Maybe we can choose a name that is less ambiguous (cf 'indexing' with embeddings for example) ?
I can think of 'reorder' (or 'rearrange', or 'apply_permutation').

Reply:
Two things here:
- I'm really happy you do this comment because that's exactly the main thing I was bothered about in this PR, this too generic name :) What an OSS connection!
- I like `reorder` let's go with this one!

Reply:
I'm hesitating with `index_gather` like in PyTorch as well. In particular I would like to emphasize that the list of indices given doesn't have to be a permutation but can be shorter than the dataset size or contain duplicated indices.

Reply:
Overall, I think the word `index` should stay FAISS/Elastic internals and not spread into the whole library. What do you think?

Reply:
Yes `index_select` is fine too !
I agree with you to keep the word `index` for internals of the Faiss/ES similarity search feature.

Reply:
Ok, let's go for `select` for the moment and see how we like it.